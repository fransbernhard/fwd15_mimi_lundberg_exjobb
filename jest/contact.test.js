import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import Contact from '../src/components/contact';

describe('components', () => {
  describe('<Contact/>', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<Contact />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

test('Input field updates on msg input', () => {
  const component = renderer.create(
    <Contact />
  );
  component
    .getInstance()
    ._handleChangeMsg({target: { value: "New msg"}});
  // re-rendering
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})


test('Input field updates on email input', () => {
  const component = renderer.create(
    <Contact />
  );
  component
    .getInstance()
    ._handleChange({target: { value: "New email"}});
  // re-rendering
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
