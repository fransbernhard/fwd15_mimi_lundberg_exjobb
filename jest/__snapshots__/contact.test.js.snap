// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field updates on email input 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value="New email"
      />
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        placeholder="meddelande"
        required={true}
        rows="8"
        value=""
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Skicka"
      />
    </form>
  </div>
</div>
`;

exports[`Input field updates on msg input 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value=""
      />
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        placeholder="meddelande"
        required={true}
        rows="8"
        value="New msg"
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Skicka"
      />
    </form>
  </div>
</div>
`;

exports[`components <Contact/> renders correctly 1`] = `
<div
  className="contact"
  id="contact"
>
  <div
    className="filter"
  >
    <form
      className="form"
      id="formContact"
      onSubmit={[Function]}
    >
      <input
        id="formEmail"
        name="formEmail"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
        value=""
      />
      <textarea
        cols="40"
        id="formMsg"
        name="formMsg"
        onChange={[Function]}
        placeholder="meddelande"
        required={true}
        rows="8"
        value=""
      />
      <input
        className="btn--cta"
        id="btn-submit"
        type="submit"
        value="Skicka"
      />
    </form>
  </div>
</div>
`;
